pype/plugins/global/publish/integrate_new.py

```diff
@@ -91,18 +91,29 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
     default_template_name = "publish"
     template_name_profiles = None
 
-    def process(self, instance):
+    # file_url : file_size of all published and uploaded files
+    integrated_file_sizes = {}
+
+    TMP_FILE_EXT = 'tmp' # suffix to denote temporary files, use without '.'
 
+    def process(self, instance):
+        self.integrated_file_sizes = {}
         if [ef for ef in self.exclude_families
                 if instance.data["family"] in ef]:
             return
 
-        self.register(instance)
-
-        self.log.info("Integrating Asset in to the database ...")
-        self.log.info("instance.data: {}".format(instance.data))
-        if instance.data.get('transfer', True):
-            self.integrate(instance)
+        try:
+            self.register(instance)
+            self.log.info("Integrated Asset in to the database ...")
+            self.log.info("instance.data: {}".format(instance.data))
+            self.handle_destination_files(self.integrated_file_sizes,
+                                          instance, 'finalize')
+        except Exception as e:
+            # clean destination
+            self.log.critical("Error when registering",  exc_info=True)
```

Comment:
I think `critical` log should be used only when log is critical for running application. This is more likely `error` level.

Reply:
Hmm, but here its really 'critical' it means that that there was some error when publishing files (either permission error, missing file, renaming step from temp files to final failed), so it means that publishing didn't provided expected files at expected location, imho.